src/sentry/api/endpoints/group_integration_details.py

```diff
@@ -264,12 +266,15 @@ def delete(self, request, group, integration_id):
                 relationship=GroupLink.Relationship.references,
             ).delete()
 
-            # check if other groups reference this external issue
-            # and delete if not
-            if not GroupLink.objects.filter(
-                linked_type=GroupLink.LinkedType.issue,
-                linked_id=external_issue_id,
-            ).exists():
-                external_issue.delete()
+            lock_key = ExternalIssue.get_delete_lock_key(external_issue.id)
+            lock = locks.get(lock_key, duration=5)
+            with TimedRetryPolicy(10)(lock.acquire):
```

Comment:
Is this too many retries? It looks like you'd use 50s in a worst case scenario. Is that longer than our web request timeout?

Reply:
good question. I just kind of looked around to see what we're doing in other places and this seemed like a common combination, although some of them are in async tasks. this one isn't though: https://github.com/getsentry/sentry/blob/master/src/sentry/api/endpoints/organization_member_index.py#L153

Reply:
Should the retries be moved into a worker task?  Do the first try in the http request and retry out of band until you get the lock?

Reply:
i don't think so. this isn't something that should take very long at all and i think moving to a task would be overcomplicating for no gain.